src/test/rpc/TransactionHistory_test.cpp

```diff
@@ -125,10 +125,10 @@ class TransactionHistory_test : public beast::unit_test::suite
             }
         }
         BEAST_EXPECT(total == 117);
-        BEAST_EXPECT(typeCounts["AccountSet"] == 20);
-        BEAST_EXPECT(typeCounts["TrustSet"] == 19);
-        BEAST_EXPECT(typeCounts["Payment"] == 58);
-        BEAST_EXPECT(typeCounts["OfferCreate"] == 20);
+        BEAST_EXPECT(typeCounts[jss::AccountSet.c_str()] == 20);
```

Comment:
Seems a little strange indexing by c_str, Is it actually doing a string comparison or is it comparing addresses? i.e. a pointer to that jss literal?

If they're all in jss then pointer lookup should be fine, but wanted to double check.

Reply:
It's doing a string comparison.  The declaration of `typeCounts` is
```
std::unordered_map<std::string, unsigned> typeCounts;
```
So we're comparing the string hash to the hash of the null-terminated c-string.  Final comparison is the built-in comparison operator between a `std::string` and a `char const*`.  Actual comparison of the pointers would probably go badly.  And, yeah, it's probably silly that an `std::unordered_map` is being used there...